src/main/java/org/apache/maven/resolver/internal/ant/types/model/MavenProject.java

```diff
@@ -0,0 +1,178 @@
+        model.setDescription(description);
+    }
+
+    /**
+     * Add a dependency to the dependencies section of the Maven project model.
+     * @param dep the Dependency object to add, must not be null
+     */
+    public void addDependency(Dependency dep) {
+        List<org.apache.maven.model.Dependency> dependencies = model.getDependencies();
+        if (dependencies == null) {
+            dependencies = new ArrayList<>();
+            model.setDependencies(dependencies);
+        }
+        org.apache.maven.model.Dependency dependency = convert(dep);
+        dependencies.add(dependency);
+    }
+
+    private org.apache.maven.model.Dependency convert(Dependency dep) {
+        org.apache.maven.model.Dependency dependency = new org.apache.maven.model.Dependency();
+        dependency.setGroupId(dep.getGroupId());
+        dependency.setArtifactId(dep.getArtifactId());
+        if (dep.getVersion() != null) {
+            dependency.setVersion(dep.getVersion());
+        }
+        if (dep.getType() != null) {
+            dependency.setType(dep.getType());
+        }
+        if (dep.getClassifier() != null) {
+            dependency.setClassifier(dep.getClassifier());
+        }
+        if (dep.getScope() != null && !dep.getScope().trim().isEmpty()) {
+            dependency.setScope(dep.getScope());
+        }
+
+        if (dep.getExclusions() != null && !dep.getExclusions().isEmpty()) {
+            dependency.setExclusions(new ArrayList<>());
+            dep.getExclusions().forEach(exclusion -> {
+                org.apache.maven.model.Exclusion mavenExclusion = new org.apache.maven.model.Exclusion();
+                mavenExclusion.setGroupId(exclusion.getGroupId());
+                mavenExclusion.setArtifactId(exclusion.getArtifactId());
+                dependency.getExclusions().add(mavenExclusion);
+            });
+        }
+        return dependency;
+    }
+
+    /**
+     * Adds a dependency to the dependency management section of the Maven project model.
+     *
+     * @param dep the Dependency object to add, must not be null
+     */
+    public void addToDependencyManagement(Dependency dep) {
+        if (model.getDependencyManagement() == null) {
+            model.setDependencyManagement(new org.apache.maven.model.DependencyManagement());
+        }
+        model.getDependencyManagement().addDependency(convert(dep));
+    }
+
+    /**
+     * Gets the Maven project model.
+     *
+     * @return the Model object representing the Maven project, never null
+     */
+    public Model getModel() {
+        return model;
+    }
+
+    /**
+     * Writes the Maven project model to the specified Writer in POM format.
+     *
+     * @param writer the Writer to write the POM to, must not be null
+     * @throws IOException if an I/O error occurs while writing
+     */
+    public void toPom(Writer writer) throws IOException {
+        MavenXpp3Writer pomWriter = new MavenXpp3Writer();
```

Comment:
I'm tempted to use an output stream as argument and then make a UTF-8 writer inside this class to guarantee the encoding.

Reply:
That probably does not have the correct encoding. Be explicit about UTF-8

Reply:
I checked MavenXpp3Writer. It uses the encoding specified in the Model:
serializer.setOutput(stream, model.getModelEncoding());
Model set encoding to UTF-8 unless explicitly overridden.

Given that it is possible to to set encoding in the org.apache.maven.model.Model, it makes sense to honor this by sticking to the OutputStream way.

Reply:
IO do argree here. You might want to leave a comment in the code for this.